### PR TITLE
Added Factory Class + Changed Difficulty Enums

### LINKED ISSUE #27: Factory for Computer Difficulty
We will have a class that acts as a factory to return the computer difficulty class based on the user's input.

### PR DESCRIPTION
**Pull request by evanugarte**
closes #27 
golly gee i sure wish it was a draft

### CODE REVIEW

File: include/setting_option.h [hunk 1]
```diff
@@ -9,17 +9,17 @@ namespace myrps
 {
 enum class SettingOption
 {
-  D_EASY,
-  D_MEDIUM,
-  D_HARD,
+  D_RANDOM,
```
Comment (drofp) [inline include/setting_option.h]: Please use `camelCased` with 'k' preface for enum naming, [per the style guide](https://google.github.io/styleguide/cppguide.html#Enumerator_Names), and [precedence set in previously named enums](https://github.com/drofp/MyRPS/blob/master/include/move.h)

**Review by drofp — CHANGES_REQUESTED** [review]
Looks pretty good! Just need to change quick naming thing

File: src/chooser_factory.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Chooser *ChooserFactory::MakeChooser(SettingOption choice)
+{
+  if(choice == SettingOption::D_RANDOM)
+  {
+    return new RandomChooser();
+  }
+  else if(choice == SettingOption::D_SMART)
+  {
+    return new MLChooser();
```
Comment (drofp) [inline src/chooser_factory.cpp]: Will likely change after merging #34, but that can be resolved in a later issue.

File: include/setting_option.h (hunk 1 above)
Comment (drofp) [reply include/setting_option.h]: done

File: src/chooser_factory.cpp (hunk 2 above)
Comment (drofp) [reply src/chooser_factory.cpp]: will resolve in #36

**Review by drofp — APPROVED** [review]

**Comment by evanugarte** [comment]
I didn't do this

**Pull request merged**